Engine/Helper.cs

```diff
@@ -700,18 +703,7 @@ public CommandInfo GetCommandInfoLegacy(string name, CommandTypes? commandType =
                 cmdletName = name;
             }
 
-            var key = new CommandLookupKey(name, commandType);
-            lock (getCommandLock)
-            {
-                if (commandInfoCache.ContainsKey(key))
-                {
-                    return commandInfoCache[key];
-                }
-
-                var commandInfo = GetCommandInfoInternal(cmdletName, commandType);
```

Comment:
If you look carefully, the `name` variable is being used for the creation of the `key` but the `cmdletname` variable is passed into this function. The `name` and `cmdletname` variable are might not always be the same but you pass only `name` into the new `GetCommandInfo()` function. I think you'd have to pass the CommandLookupKey into it instead to preserve legacy behaviour. We have seen it very hard to make any change to this method without breaking something, hence why this method is marked as legacy.